### PR TITLE
Document `base_types.py`

### PR DESCRIPTION
**Pull request by SamWilsn**
This PR removes some unnecessary constants, renames some classes for consistency, and updates `base_types.py` to markdown.

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: src/ethereum/base_types.py [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+sequences containing an exact number of bytes.
+
+[`Uint`]: ref:ethereum.base_types.Uint
+[`FixedUint`]: ref:ethereum.base_types.FixedUint
+[`U32`]: ref:ethereum.base_types.U32
+[`U256`]: ref:ethereum.base_types.U256
+[`Bytes`]: ref:ethereum.base_types.Bytes
+[`FixedBytes`]: ref:ethereum.base_types.FixedBytes
+[`Bytes0`]: ref:ethereum.base_types.Bytes0
+[`Bytes64`]: ref:ethereum.base_types.Bytes64
```
Comment (gurukamath) [inline src/ethereum/base_types.py:19]:
```suggestion
[`Uint`]: ref:ethereum.base_types.Uint
[`FixedUint`]: ref:ethereum.base_types.FixedUint
[`U32`]: ref:ethereum.base_types.U32
[`U64`]: ref:ethereum.base_types.U64
[`U256`]: ref:ethereum.base_types.U256
[`Bytes`]: ref:ethereum.base_types.Bytes
[`FixedBytes`]: ref:ethereum.base_types.FixedBytes
[`Bytes0`]: ref:ethereum.base_types.Bytes0
[`Bytes4`]: ref:ethereum.base_types.Bytes4
[`Bytes8`]: ref:ethereum.base_types.Bytes8
[`Bytes20`]: ref:ethereum.base_types.Bytes20
[`Bytes32`]: ref:ethereum.base_types.Bytes32
[`Bytes64`]: ref:ethereum.base_types.Bytes64
[`Bytes256`]: ref:ethereum.base_types.Bytes256
```

File: tests/helpers/load_difficulty_tests.py [hunk 2]
```diff
@@ -1,19 +1,9 @@
...
-from ethereum.base_types import U256, Uint
-from ethereum.utils.hexadecimal import (
-    hex_to_bytes,
-    hex_to_bytes8,
-    hex_to_bytes32,
-    hex_to_hash,
-    hex_to_u256,
-    hex_to_uint,
-)
+from ethereum.utils.hexadecimal import hex_to_u256, hex_to_uint
```
Comment (gurukamath) [inline tests/helpers/load_difficulty_tests.py:6]: Interesting that `flake8` did not catch this. Will create an issue to find more such cases

**Review by gurukamath — CHANGES_REQUESTED** [review]

File: src/ethereum/base_types.py (hunk 1 above)
Comment (SamWilsn) [reply src/ethereum/base_types.py:19]: These are markdown [reference links](https://spec.commonmark.org/0.31.2/#reference-link), and are only necessary if a link exists in the text.

Unless you're suggesting that I also add all the types to the text?

Comment (gurukamath) [reply src/ethereum/base_types.py:19]: Ah. Got it

**Review by gurukamath — APPROVED** [review]

**Review by petertdavies — APPROVED** [review]

**Pull request merged**